infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentManager.java

```diff
@@ -47,10 +103,87 @@
         return segmentCopy
                 .values()
                 .stream()
-                .sorted(Comparator.comparingLong(StreamLogSegment::getStartAddress))
+                .sorted(Comparator.comparingLong(StreamLogSegment::getSegment))
                 .limit(segmentCopy.size() - logParams.protectedSegments)
                 .collect(Collectors.toList());
     }
 
-    // TODO: Merge method is merge is needed.
+    /**
+     * Prune the provided segments, close and remove their reference in the maps to
+     * save space and improve lookup performance. Usually the provided segments are
+     * relatively old and are less likely to be accessed later.
+     *
+     * @param segments segments to prune
+     */
+    public void pruneSegmentMapping(List<StreamLogSegment> segments) {
+        segments.forEach(segment -> {
+            segment.close();
+            streamLogSegments.remove(segment.getSegment());
+        });
+    }
+
+    /**
+     * Only for prefix trim, will be removed.
+     */
+    public void cleanAndClose(long endSegment) {
+        cleanAndClose(streamLogSegments, endSegment);
+        cleanAndClose(garbageLogSegments, endSegment);
+
+        deleteFiles(endSegment);
+    }
+
+    private void cleanAndClose(Map<Long, ? extends AbstractLogSegment> segmentMap, long endSegment) {
+        List<AbstractLogSegment> segments = segmentMap.values()
+                .stream()
+                .filter(segment -> segment.getSegment() <= endSegment)
+                .collect(Collectors.toList());
+
+        segments.forEach(segment -> {
+            segment.close();
+            segmentMap.remove(segment.getSegment());
+        });
+    }
+
+    private void deleteFiles(long endSegment) {
+        int numFiles = 0;
+        long freedBytes = 0;
+        File dir = logDir.toFile();
+        File[] files = dir.listFiles(file -> {
+            try {
+                String segmentNum = file.getName().split("\\.")[0];
+                return Long.parseLong(segmentNum) < endSegment;
+            } catch (Exception e) {
+                log.warn("trimPrefix: ignoring file {}", file.getName());
+                return false;
```

Comment:
the return type is void.

Reply:
This is inside lambda, so a boolean is needed.